### DISCUSSION TITLE
Need Help What do I do? Step 4 in runme PS1

### DISCUSSION THREAD

**Discussion by BoydAcres** [MRColorR/money4band#11] [Q&A] [answered]
This menu item will launch all the apps using the configured .env file and the docker-compose.yml file (Docker must be already installed and running)
Do you wish to proceed Y/N?: y
**docker : The term 'docker' is not recognized as the name of a cmdlet, function, script file, or operable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At C:\Users\VALKYRIE\Downloads\money4band-v.1.2.7\money4band-v.1.2.7\runme.ps1:339 char:9
+         docker compose up -d
+         ~~~~~~
    + CategoryInfo          : ObjectNotFound: (docker:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException**

All Apps started you can visit the web dashboard on http://localhost:8081/ . If not already done use the previously generated earnapp node URL to add your device in your earnapp dashboard.
Check the README file for more details. Now press enter to go back to the menu:

**Comment by MRColorR** [comment]
Hello and thanks to reaching out to me, your feedback is very important.
Are you on Windows? On wich version?
Your issue seems related with docker executable path non registered as a system env variable. after installing docker a reboot is needed to update this system informations.

Please, could you one a powershell and issue this command: `docker ps` to see if docker is installed and running .

**Comment by BoydAcres** [comment]
Hello ty for your reply, i am on windows10 pro 64 bit 21h2 . attached is a screenshot, docker on showing in the background with no program loaded. 

![Screenshot 2022-11-20 090839](https://user-images.githubusercontent.com/118641873/202906797-6cdbcd1e-73c1-40bd-861f-6749a01a1f7b.png)

Windows PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.

Try the new cross-platform PowerShell https://aka.ms/pscore6

PS C:\Users\VALKYRIE> docker ps
docker : The term 'docker' is not recognized as the name of a cmdlet, function, script file, or operable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
At line:1 char:1
+ docker ps
+ ~~~~~~
    + CategoryInfo          : ObjectNotFound: (docker:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

PS C:\Users\VALKYRIE>


TY for your help. i followed you on your videos i hope we can get this running and start making so dough

**Reply by BoydAcres** [reply]
upgraded powershell to 7.3.0 same results.

**Reply by BoydAcres** [reply]
![Screenshot 2022-11-20 093257](https://user-images.githubusercontent.com/118641873/202908124-8bf5ab80-8cdd-431d-857e-6de7c63606c9.png)
 i also just now discovered that i needed to turn on windows features:
Containers and hyper V

**Reply by BoydAcres** [reply]
PowerShell 7.3.0
PS C:\Users\VALKYRIE\AppData\Local\Temp\Rar$EXa12976.38079> docker ps
docker: The term 'docker' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
PS C:\Users\VALKYRIE\AppData\Local\Temp\Rar$EXa12976.38079>

Still same thing.

**Reply by BoydAcres** [reply]
Trying this resolution:  https://forums.docker.com/t/the-term-docker-is-not-recognized-as-the-name-of-a-cmdlet-function-script-file-or-operable-program/107415

**Reply by BoydAcres** [reply]
back to step 4 on runme.ps1.... (hyper V is turned back off because of above thread)
![Screenshot 2022-11-20 111834](https://user-images.githubusercontent.com/118641873/202913334-d6f4be06-8c7d-497a-aaff-29372ba26af0.png)
now im stuck here.

**Reply by MRColorR** [reply]
> back to step 4 on runme.ps1.... (hyper V is turned back off because of above thread) ![Screenshot 2022-11-20 111834](https://user-images.githubusercontent.com/118641873/202913334-d6f4be06-8c7d-497a-aaff-29372ba26af0.png) now im stuck here.

the project seems to work fine, it's the docker installation that is behaving strangely. Also the error you had initially (also the second one) depends on the docker installation not the project setup. Did you do the manual docker install or the automatic one provided in the project script?

**Reply by MRColorR** [reply]
> Trying this resolution: https://forums.docker.com/t/the-term-docker-is-not-recognized-as-the-name-of-a-cmdlet-function-script-file-or-operable-program/107415

just an heads up about this: be careful, it fix this docker installation issue regarding commands not runnig caused by the docker PATH variable not being added to system variables,  but the linked solution is for windows server. It may use (and often uses in many guides) a different than the standard docker installation system which while it shouldn't prevent the use of this project or other docker containers, can cause some strange behavior on windows home or pro. Better to install docker following the directions on the official docker site or use the unattended installation provided in the project with the script that automates the standard installation of docker desktop.

**Comment by BoydAcres** [comment]
I believe i am resolved: 
![image](https://user-images.githubusercontent.com/118641873/202915999-005d7ff2-95ee-4237-93e2-e63fc82bd313.png)
any way to check or verify this further. what's the next step if any?

**Reply by MRColorR** [reply]
> I believe i am resolved: ![image](https://user-images.githubusercontent.com/118641873/202915999-005d7ff2-95ee-4237-93e2-e63fc82bd313.png) any way to check or verify this further. what's the next step if any?

there you are just running the bitping setup container. the steps to use this project are to setup the env file and then run the stack. in docker desktop you will then see all the containers gathered in a dropdown menu.

**Comment by BoydAcres** [comment]
automatic. now i have had to do a reboot and i lost the container. so now i
have to add it again.

**Reply by MRColorR** [reply]
that container is ephemeral and it is fine that it disappears. is launched during the setup of the env file. the complete stack is composed of multiples containers

**Comment by MRColorR** [comment]
how did you install docker ? I think your docker installation is incomplete or corrupted. Try uninstalling docker and reinstalling using option 2 in the menu or following the instructions on the docker desktop site.

**Reply by BoydAcres** [reply]
doing now.

**Reply by BoydAcres** [reply]
Can you tell me if I need to enable Hyper V? the above forum I saw convinced me to turn off this feature, I'm curious if you think I need to enable?

**Reply by MRColorR** [reply]
hyper-v is not needed to use docker. Just enable virtualization platform and wsl features. Also install wsl update package or use the wsl on the Microsoft store.

**Comment by BoydAcres** [comment]
ok reinstalled with auto step 2. 
completed step 3. 
![image](https://user-images.githubusercontent.com/118641873/202925052-e0951c14-6257-4945-a49a-158b462b4565.png)
now to step 4.

**Comment by BoydAcres** [comment]
this is what i get from running option 4:

This menu item will launch all the apps using the configured .env file and the docker-compose.yml file (Docker must be already installed and running)
Do you wish to proceed Y/N?: y
[+] Running 0/0
 - watchtower Pulling                                                                                                                            0.0s
 - earnapp Pulling                                                                                                                               0.0s
 - honeygain Pulling                                                                                                                             0.0s
 - Peer2profit Pulling                                                                                                                           0.0s
 - Repocket Pulling                                                                                                                              0.0s
 - Packetstream Pulling                                                                                                                          0.0s
 - TraffMonetizer Pulling                                                                                                                        0.0s
 - iproyal Pulling                                                                                                                               0.0s
 - webdashboard Pulling                                                                                                                          0.0s
error getting credentials - err: docker-credential-desktop resolves to executable in current directory (.\docker-credential-desktop), out: ``
All Apps started you can visit the web dashboard on http://localhost:8081/ . If not already done use the previously generated earnapp node URL to add your device in your earnapp dashboard.
Check the README file for more details. Now press enter to go back to the menu:


what do i do?

**Reply by MRColorR** [reply]
thanks for the info, i'll look into this now

**Reply by MRColorR** [reply]
it's related to the docker credentials. This is a docker feature not used in this project as all files and images used could be downloaded from docker default register without authentication. The simplest solution to attempt is to try to reboot your machine to force your account informations and credentials to refresh, and then start the stack again opening a powershell in the project root folder and using `docker compose up -d`

Edit: if this is not enough look to this post https://forums.docker.com/t/docker-credential-desktop-exe-executable-file-not-found-in-path-using-wsl2/100225/30 that explains why this happens (e.g. in case like: had a bad docker installation then removed it but leftovers config files are causing problems with the new correct installation).

In a nutshell the docker config file is not correct due to an old bad docker installation that left some residue after uninstall, docker tries to read and use some credentials that don't exist for downloading the requested images. You can proceed to eliminate this incorrect configuration by deleting the following portion from the docker configuration file (be careful to keep the indentation and the right parentheses:
`{
	"auths": {
		"docker.your.domain.": {
			"auth": "badcredentials1234="
		}
	}
}`
the file to "fix" is located in the path:
  - in Windows:
C:\Users\[YOUR.USERNAME]\.docker\config.json
- in linux $HOME/.docker/config.json

**Comment by MRColorR** [comment, marked as answer]
In a nutshell the docker config file is not correct due to an old bad docker installation that left some residue after uninstall, docker tries to read and use some credentials that don't exist for downloading the requested images. 
Fix you could try:
- Reboot your machine to force a credentials update for your account , this fix the issue if it's caused from a not refreshed account info and  permissions
- Delete leftovers incorrect configuration from the .docker/config.json file removing `{ "auths": { "docker.your.domain.": { "auth": "badcredentials1234=" } } }` from the file, this fix the issue if it's caused by a previous bad docker installation or misconfiguration
- Delete the entire Docker for desktop configuration file in  .docker/config.json as if used it will be regenerated by docker.
  - in Windows the file is located under C:\Users\yourUserName\\.docker\config.json
  - in linux the file is located under $HOME/.docker/config.json
- Do a Docker complete uninstall and clean reinstall, (e.g if on windows use a tool like [Revo uninstaller portable](https://www.revouninstaller.com/it/revo-uninstaller-free-download/) and use it to completely uninstall docker and delete also all the leftovers found by the scan). Also check that there are no docker folders left in paths like :
** On windows C:\Program Files , C:\Program Files(x86), %AppData% (or C:\Users\yourusername\AppData\Roaming) and C:\Users\[YOUR.USERNAME]\
** On linux after purging the docker installation from the system search also under $HOME/
** if present, delete all residual docker folders and the files inside them
** Now reinstall docker
